pkg/controller/trainjob_controller.go

```diff
@@ -83,54 +75,70 @@ func (r *TrainJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	var err error
+	// Keep track of the origin TrainJob status
+	originStatus := trainJob.Status.DeepCopy()
+
+	// Let's clear the failed condition that could have been set previously.
+	// An external change to the TrainJob spec may transition it out of the Failed state.
+	removeFailedCondition(&trainJob)
+
 	runtimeRefGK := jobruntimes.RuntimeRefToRuntimeRegistryKey(trainJob.Spec.RuntimeRef)
 	runtime, ok := r.runtimes[runtimeRefGK]
 	if !ok {
-		return ctrl.Result{}, fmt.Errorf("unsupported runtime: %s", runtimeRefGK)
+		err = fmt.Errorf("unsupported runtime: %s", runtimeRefGK)
+		setFailedCondition(&trainJob, fmt.Sprintf("unsupported runtime: %s", runtimeRefGK), trainer.TrainJobRuntimeNotSupportedReason)
+	} else {
+		err = r.reconcileObjects(ctx, runtime, &trainJob)
+		if err != nil {
+			// TODO (astefanutti): the error should be surfaced in the TrainJob status to indicate
+			//  the creation of the runtime resources failed and the TrainJob is backed off until
+			//  the next retry attempt.
+			r.recorder.Eventf(&trainJob, corev1.EventTypeWarning, "TrainJobReconcileFailed",
```

Comment:
I would also name it as `TrainJobResourcesCreationFailed` to not confuse users since reconciliation of TrainJob happens even after the TrainJob's resources are deployed.
```suggestion
			r.recorder.Eventf(&trainJob, corev1.EventTypeWarning, "TrainJobResourcesCreationFailed",
```

Reply:
Sounds good to me. Updated according to your suggestion too.